pkg/api/mock/mock_api_implementation.go

```diff
@@ -0,0 +1,215 @@
+	}
+
+	if options.Paged {
+		if options.Page == 0 {
+			options.Page = 1
+		}
+
+		pageInfo, err := newMockPageIter(objectsWithSameType, options.EntriesPerPage, options.Page)
+		if err != nil {
+			return err
+		}
+		*options.PageInfo = pageInfo
+	}
+
+	if options.ObjectChannel != nil {
+		c := make(chan types.ObjectRetriever)
+		*options.ObjectChannel = c
+
+		objectRetrieved := make(chan bool)
+		go func(pi types.PageInfo) {
+			var pageData []types.Object
+
+		outer:
+			for pi.Next(&pageData) {
+				for _, o := range pageData {
+					// since we are in a goroutine, we might already be in the next iteration of this loop
+					// at the time the receiving end of this channel calls the closure. Having a loop-body
+					// scoped variables makes the data for the closure perfectly identified.
+					closureData := o
+					c <- func(out types.Object) error {
+						reflect.ValueOf(out).Elem().Set(reflect.ValueOf(closureData).Elem())
+
+						select {
+						case <-ctx.Done():
+						case objectRetrieved <- true:
+						}
+
+						return nil
+					}
+
+					select {
+					case <-ctx.Done():
+						break outer
+					case <-objectRetrieved:
+					}
+				}
+			}
+
+			close(c)
+		}(channelPageIterator)
+	}
+
+	return nil
+}
+
+// Saves an types.Object to the MockAPIs local storage and sets a RandomIdentifier
+// if not already present. No validation is performed!
+func (a mockAPI) Create(ctx context.Context, o types.Object, opts ...types.CreateOption) error {
+	if rwt, ok := o.(*corev1.ResourceWithTag); ok {
+		if obj, ok := a[rwt.Identifier]; ok {
+			obj.Tags[rwt.Tag] = true
+		} else {
+			return api.ErrNotFound
+		}
+		return nil
+	}
+
+	identifier := test.RandomIdentifier()
+	v := reflect.Indirect(reflect.ValueOf(o))
+	t := reflect.TypeOf(o).Elem()
+	for i := 0; i < t.NumField(); i++ {
+		if tag, ok := t.Field(i).Tag.Lookup("anxcloud"); ok &&
+			tag == "identifier" &&
+			t.Field(i).Type.Kind() == reflect.String &&
+			v.Field(i).String() == "" {
+			v.Field(i).SetString(identifier)
+		}
+	}
+
+	identifier, err := api.GetObjectIdentifier(o, true)
+	if err != nil {
+		return err
+	}
+
+	cloned, err := copystructure.Copy(o)
+	if err != nil {
+		return err
+	}
+
+	a[identifier] = &mockAPIObject{Object: cloned.(types.Object), Tags: make(map[string]bool)}
+
+	return nil
+}
+
+// Overwrites a types.Object in MockAPIs local storage
+func (a mockAPI) Update(ctx context.Context, o types.IdentifiedObject, opts ...types.UpdateOption) error {
+	identifier, err := api.GetObjectIdentifier(o, false)
+	if err != nil {
+		return err
+	}
+
+	if val, ok := a[identifier]; ok && !val.Deleted {
+		cloned, err := copystructure.Copy(o)
+		if err != nil {
+			return err
+		}
+		val.Object = cloned.(types.Object)
+	} else {
+		return api.ErrNotFound
+	}
+
+	return nil
+}
+
+// Removes a types.Object from MockAPIs local storage
+func (a mockAPI) Destroy(ctx context.Context, o types.IdentifiedObject, opts ...types.DestroyOption) error {
```

Comment:
@LittleFox94 do you have any advice on how to solve this?

Reply:
having a `(a mockAPI) untag(o types.IdentifiedObject) (isTag bool, err error)` and call it like this would change two returns to only one:
```go
if isTag, err := a.untag(o); isTag {
  return err
}
```

something similar for `a.tag()` would help you for `Create`, too.